files/build_templates/docker_image_ctl.j2

```diff
@@ -5,6 +5,15 @@ function getMountPoint()
     echo $1 | python -c "import sys, json, os; mnts = [x for x in json.load(sys.stdin)[0]['Mounts'] if x['Destination'] == '/usr/share/sonic/hwsku']; print '' if len(mnts) == 0 else os.path.basename(mnts[0]['Source'])" 2>/dev/null
 }
 
+function updateHostName()
+{
+    HOSTNAME=`hostname`
+
+    echo "Set hostname in {{docker_container_name}} container"
+    docker exec -i {{docker_container_name}} bash -c "hostname $HOSTNAME"
```

Comment:
>docker exec -i {{docker_container_name}} bash -c "hostname $HOSTNAME" [](start = 4, length = 69)

It is possible that the early started process see the original hostname.

Reply:
i've checked this using 'dhcp_relay' testcase which is required the hostname to pre-set to a known state and it's working. Anyway, it's better than the existing state as we'll keep hostname to be in sync w/ base system. 
In addition - as for my understanding, there is no way to set some specific hostname while starting the container. the only way to set a proper hostname for all processes - remove the previous instance of a container each time.. but this could bring timing issue.

Reply:
This is the comparison between your PR and https://github.com/Azure/sonic-buildimage/pull/2773. That one has limited scope on only one container, and don't allow early early processes get the original hostname, because the processes order are controlled by start.sh.

This proposed PR are meant to be general, but has this disadvantage. So it is not qualified for a complete solution.

If you require any fix on specific container, you may follow that PR and fix in the same way.